core/src/main/java/hudson/cli/ReloadConfigurationCommand.java

```diff
@@ -43,8 +46,26 @@ public String getShortDescription() {
 
     @Override
     protected int run() throws Exception {
-        Jenkins.getActiveInstance().doReload();
-        return 0;
+        Jenkins j = Jenkins.getInstance();
+        // Or perhaps simpler to inline the thread body of doReload?
+        j.doReload();
+        Object app;
+        while ((app = WebApp.get(j.servletContext).getApp()) instanceof HudsonIsLoading) {
+            Thread.sleep(100);
```

Comment:
Should be a timeout there? I know cases when Jenkins infinitely hangs on reloading. Not sure it is safe to just leave it as is then